diesel/src/mysql/connection/raw.rs

```diff
@@ -0,0 +1,97 @@
+extern crate mysqlclient_sys as ffi;
+
+use std::ffi::CStr;
+use std::os::{raw as libc};
+use std::ptr;
+use std::sync::{Once, ONCE_INIT};
+
+use result::{ConnectionResult, ConnectionError};
+use super::url::ConnectionOptions;
+
+pub struct RawConnection(*mut ffi::MYSQL);
+
+impl RawConnection {
+    pub fn new() -> Self {
+        perform_thread_unsafe_library_initialization();
+        let raw_connection = unsafe { ffi::mysql_init(ptr::null_mut()) };
+        if raw_connection.is_null() {
+            panic!("Insufficient memory to allocate connection");
+        }
+        let result = RawConnection(raw_connection);
+
+        // This is only non-zero for unrecognized options, which should never happen.
+        let charset_result = unsafe { ffi::mysql_options(
+            result.0,
+            ffi::mysql_option::MYSQL_SET_CHARSET_NAME,
+            b"utf8mb4\0".as_ptr() as *const libc::c_void,
```

Comment:
Don't get me started on mysql utf8 support… a coworker recently had to migrate their Redmine instance to utf8mb4 (that's "UTF-8, much better, fourth try", right?), because I used an emoji in a commit message.

The docs [say](https://dev.mysql.com/doc/refman/5.7/en/mysql-options.html),

> The name of the character set to use as the default character set.

What does setting the default charset actually do? Does mysql use that when you don't specify the charset in a `create table` or `create database` call? Or some more magic?